core/src/main/java/org/infinispan/Version.java

```diff
@@ -24,25 +26,39 @@
 
    private static final Version INSTANCE = new Version();
 
+   private final String version;
+   private final String codename;
    private final byte[] versionId;
    private final String moduleSlot;
    private final short versionShort;
    private final short marshallVersion;
    private final String majorMinor;
    private final String major;
 
+
    private Version() {
-      String parts[] = getParts(Injected.getVersion());
+      Properties properties = new Properties();
+      try (InputStream is = Version.class.getResourceAsStream("/META-INF/infinispan-version.properties")) {
```

Comment:
I guess the "safe" way is to ensure that we are getting the resource from the same root path as from where the Version class was loaded. Or wait for javassist to fix their issue.